pkg/config/config.go

```diff
@@ -33,7 +29,26 @@ func Load(config *api.Config) error {
 		klog.Warning("'target.repoName' property is empty. Using 'myrepo' default value")
 		config.Target.RepoName = defaultRepoName
 	}
-
+	if config.Source.Repo.Auth == nil {
+		sourceAuth := &api.Auth{}
+		if su := os.Getenv("SOURCE_AUTH_USERNAME"); su != "" {
+			sourceAuth.Username = su
+		}
+		if sp := os.Getenv("SOURCE_AUTH_PASSWORD"); sp != "" {
+			sourceAuth.Password = sp
+		}
+		config.Source.Repo.Auth = sourceAuth
```

Comment:
A quick comment that apply to the other block:

Use short variables when it makes sense (like here). The [basic rule](https://github.com/golang/go/wiki/CodeReviewComments#variable-names): the further from its declaration that a name is used, the more descriptive the name must be.

Hence, you can use `u` and `p` to refer to `username` and `password` in the `if-Getenv` scope and `a` (or `auth`) to refer to `api.Auth` in the `if-Auth` scope.

Another approach:

```suggestion
		config.Source.Repo.Auth := &api.Auth{}
		if u := os.Getenv("SOURCE_AUTH_USERNAME"); u != "" {
			config.Source.Repo.Auth.Username = u
		}
		if p := os.Getenv("SOURCE_AUTH_PASSWORD"); p != "" {
			config.Source.Repo.Auth.Password = p
		}
```

Another point is, does it make sense to provide username without a password? Otherwise, I would do a double-check:

```suggestion
		if u, p := os.Getenv("SOURCE_AUTH_USERNAME"), os.Getenv("SOURCE_AUTH_PASSWORD"); u != "" && p != "" {
			config.Source.Repo.Auth = &api.Auth{Username: u, Password: p}
        }
```

And finally, you could even add a method to the `api` package to set the password, so you don't have to duplicate it:

```
# api.go

func (r *Repo) SetBasicAuth(user string, pass string) error {
  // No auth provided, leave as-is
  if user == "" && pass == "" {
      return nil
  }
  // TODO: must both be set? then manage each case

  r.Auth = &Auth{Username: user, Password: pass}
  return nil
}
```

so you can simplify the code to:

```
	if config.Target.Repo.Auth == nil {
        if err := config.Target.Repo.SetBasicAuth(os.Getenv("TARGET_AUTH_USERNAME"), os.Getenv("TARGET_AUTH_PASSWORD")); err != nil {
            return err
        }
    }
```

However, I am just mentioning these options to you. Another option is to leave the code as-is right now if you prefer.